File: python/cup_cakes.py
1.
def find_smallest(numbers):
  
  if not numbers:
    return None

  smallest = numbers[0]

  for number in numbers:
    if number < smallest:
      smallest = number

  return smallest


print(f"The smallest number is: {smallest(number)}")average array(int[]

2.

def averageArray(numbers):

    if not numbers:
        return 0.0
    
    return sum(numbers) / len(numbers)

print(f"The average is: {averageArray(my_list)}")

3.

def countOccurence(number, target):
   
    count = 0
    
    
    for element in arr:
        
        if element == target:
           
            count += 1
            
    return count

    numbers = [1, 5, 3, 5, 2, 5, 8, 5]
    target = 5

print(f"The number {target} appears {countOccurence(numbers, target)} times.")

4.

def containsElement(array, target):
 
  return target in array
  numbers = [10, 20, 30, 40, 50]

  firsttarget = 30
  print(f"Does the list contain {target1}? {containsElement(numbers, target1)}")

  secondtarget = 99

print(f"Does the list contain {target2}? {containsElement(numbers, target2)}")


5.
def get_first_element(array):

	if len(array) == 0:
        return 0

print(get_last_element([10, 20, 30]))
print(get_last_element([]))


6.
def get_last_element(array):
    if len(arr) == 0:
        return 0
    return arr[-1]

print(get_last_element([10, 20, 30]))
print(get_last_element([]))


7.

def count_elements(array):
    count = 0
    for _ in array:
        count += 1
    return count

print(count_elements([10, 20, 30, 40])) 
print(count_elements([]))

8.

def get_middle_element(array):

    if not array:
        return None 

    middle_index = len(array) // 2

    if len(array) % 2 == 0:
        return array[middle_index - 1]  
    else:
        return array[middle_index]  

print(get_middle_element([10, 20, 30]))       
print(get_middle_element([10, 20, 30, 40]))     
print(get_middle_element([100]))            
print(get_middle_element([]))              

9.

def swap_first_and_last(array):
    if len(arr) < 2:
        return arr[:]  

    result = arr[:]  
    result[0], result[-1] = result[-1], result[0]  
    return result


	original = [10, 20, 30, 40, 50]
	swapped = swap_first_and_last(original)

print("Original:", original)
print("Swapped: ", swapped)

File: java/CupCakes.java
1.

public class CupCakes {


	public static int Smallest(int[] number) {

		if (int number = 0; number.length = 0) {

                System.out.println("The array is empty.");
   
            return Integer.MAX_VALUE;
        }

                int smallest = number[0];

        for (int i = 1; i < number.length; i++) {

                if (number[i] < smallest) {
               
                    smallest = number[i];
            }
        }

        return smallest;
    }


2.

public static double averageArray(int[] arr) {
        
        if (number == length == 0) {
            return 0;
        }

        double sum = 0;
        
        for (int element: number) {
            sum += element;
        }
        return sum / number.length;
    }
    
3.

public static int countOccurence(int[] number, int target) {

        int count = 0;

        for (int element: number) {
       
            if (element == target) {
                count++;
            }
        }

        return count;
    }


4.

public static boolean containsElement(int[] number, int target) {
        
        for (int element: number) {
           
            if (element == target) {
  
                return true;
            }
        }
        return false;
    }

5.

public static int getFirstElement(int[] arr) {

        if (array == null || array.length == 0) {
            return 0;
        }
        return array[0];


6.

public static int getLastElement(int[] arr) {
        if (arr == null || arr.length == 0) {
            return 0;
        }
        return arr[arr.length - 1];
    }

7.

public static int countElements(int[] arr) {
        int count = 0;
        try {
            while (true) {
                int temp = arr[count];
                count++;
            }
        } catch (ArrayIndexOutOfBoundsException e) {

        }
        return count;
    }

8.

public static Integer getMiddleElement(int[] arr) {
        if (arr == null || arr.length == 0) {
            return null;
        }
        int midIndex = (arr.length - 1) / 2;
        return arr[midIndex];
    }




9.

public static int[] swapFirstAndLast(int[] arr) {
        if (arr == null || arr.length < 2) {
      
            return arr;
        }

        int[] result = arr.clone();

        int temp = result[0];
        result[0] = result[result.length - 1];
        result[result.length - 1] = temp;

        return result;
    }


